www/js/survey/enketo-survey.js

```diff
@@ -206,6 +207,7 @@ angular.module('emission.survey.enketo.launch', [
         console.log("Added " + ClientStats.getStatKeys().BUTTON_FORCE_SYNC + " event");
       });
     return ControlSyncHelper.forceSync().then(function () {
+      console.log('forceSync start');
```

Comment:
This line does not get executed.

Reply:
Unfortunately, that is not a valid assumption.

The sync code was originally designed to be only automatically triggered so that the data would sync periodically. I later added an interface to force the sync, but that just triggers the sync process. So `forceSync` triggers the sync, but you can't rely on knowing when the sync is complete.

There were some prior issues in which I debugged this in detail. Did you check through prior issues/code history to validate your assumption?

Reply:
> Unfortunately, that is not a valid assumption. There were some prior issues in which I debugged this in detail. Did you check through prior issues/code history to validate your assumption?
> 
> IIRC, the sync code was originally designed to be only automatically triggered so that the data would sync periodically. I later added an interface to force the sync, but that just triggers the sync process. So `forceSync` triggers the sync, but you can't rely on knowing when the sync is complete.

How does this any different than the force sync button in the settings.js page though? Or does this mean devapp and emTripLog behave differently for the sync process?

Reply:
@shankari so speaking on the high-level, what is your suggestion if we want to make sure the end survey get sent right away?

Reply:
It is not any different from the `forceSync` button in the `settings.js` page. The sync is launched and completes. It just doesn't trigger the `then` callback.

@asiripanich note that we only sync values before the current end of trip. You want to use the `endTripAndForceSync` method to push the end survey.  The sync will still happen, but you won't be able to display a popup when the sync is complete since the callback is not triggered. Please see the history around this code for more details.